File: ios/MyReactNativeApp/MyNativeModule.m
//
//  MyNativeModule.m
//  MyReactNativeApp
//
//  Created by fauquette fred on 26/01/18.
//  Copyright © 2018 Fred Fauquette. All rights reserved.
//

#import <Foundation/Foundation.h>

#import <React/RCTBridgeModule.h>

@interface MyNativeModule : NSObject
@end

@interface MyNativeModule (RCTExternModule) <RCTBridgeModule>
@end

@implementation MyNativeModule (RCTExternModule)
dispatch_queue_t RCTJSThread;
void RCTRegisterModule(Class);
+ (NSString *)moduleName {
    return @"MyNativeModule";
}
+ (void)load {
    RCTRegisterModule(self);
}

+ (const RCTMethodInfo *)__rct_export__uniqueID {
    static RCTMethodInfo config = {"triggerJSRequest", "triggerRequest", NO};
    return &config;
}

@end

File: ios/MyReactNativeApp/MyNativeModule.swift
//
//  MyNativeModule.swift
//  MyReactNativeApp
//
//  Created by fauquette fred on 26/01/18.
//  Copyright © 2018 Fred Fauquette. All rights reserved.
//

import Foundation
import React


@objc(MyNativeModule)
class MyNativeModule: NSObject {
    override init() {
        super.init()
    }
    @objc static func requiresMainQueueSetup() -> Bool {
        return false
    }
    @objc var bridge: RCTBridge?
    @objc var methodQueue: DispatchQueue?
    @objc func triggerRequest() -> Void {
        print("trigger Request")
    }
    @objc func batchDidComplete() {
        // do something if needed at the end of the batch
    }
    @objc func partialBatchDidFlush() {
        // do something if needed at the end of the partial flush of the batch
    }
}

